core/payment.go

```diff
@@ -179,67 +179,71 @@ func (p *PaymentSystem) checkAmount(accountUUID uuid.UUID, amount uint) error {
 }
 
 func (p *PaymentSystem) GetAccounts(userUUID uuid.UUID) ([]models.Account, error) {
-	return p.UserRepo.GetAccountsForUser(userUUID)
+	return p.Repo.GetAccountsForUser(userUUID)
 }
 
 func (p *PaymentSystem) GetTransactions(accountUUID uuid.UUID) ([]models.Transaction, error) {
-	return p.UserRepo.GetTransactionForAccount(accountUUID)
+	return p.Repo.GetTransactionForAccount(accountUUID)
 }
 
 func (p *PaymentSystem) SendTransaction(transactionUUID uuid.UUID) (models.Transaction, error) {
-	transaction, err := p.UserRepo.GetTransactionByUUID(transactionUUID)
+	err := p.Repo.Transaction(
+		func(repo repository.Repository) error {
+			transaction, err := p.Repo.GetTransactionByUUID(transactionUUID)
```

Comment:
Я забув сказати і не помітив одразу. Отут тепер треба використовувати `repo`, а не `p.Repo`:
```suggestion
			transaction, err := repo.GetTransactionByUUID(transactionUUID)
```